nifi-stateless/nifi-stateless-core/src/main/java/org/apache/nifi/stateless/core/StatelessFlow.java

```diff
@@ -99,6 +122,194 @@ public StatelessFlow(final List<StatelessComponent> roots) {
         this.componentFactory = null;
     }
 
+    public StatelessFlow(final ProcessGroupDTO flow, final ExtensionManager extensionManager, final VariableRegistry variableRegistry, final List<String> failureOutputPorts,
```

Comment:
Both the 2 constructors and the 2 `create...AndEnque...` methods could be normalized with a common private method.

Reply:
I'm not sure how I should approach this one. I started with using the strategy pattern to get around the sleight differences between `Versioned*` objects and `*DTO` objects but I feel like I ended up with too many generics for what I'm trying to achieve. Do you have any recommendations for how to refactor this?

Reply:
Added a new compatibility package to reconcile the differences between these classes and eliminate the code duplication